app/components/bali/calendar/header/component.rb

```diff
@@ -39,6 +39,12 @@ def next_start_date
           end
         end
 
+        def route(params = {})
+          uri = URI.parse(route_path)
+          uri.query = params.merge!(CGI.parse(uri.query.to_s)).to_query
+          uri.to_s
+        end
```

Comment:
Creo que esto lo podemos dividir en un par de métodos para no tener que hacer el parsing cada vez que llamamos el método **route**. (lo usamos 4 veces)

```suggestion
        def uri
          @uri ||= URI.parse(route_path)
        end
        
        def query_params
          @query_params ||= CGI.parse(uri.query.to_s)
        end
        
        def route(params = {})
          uri.query = query_params.merge(params).to_query # Aqui hay que hacer el merge al revés, para que los params que le pasas como argumento tomen precedencia sobre los que ya vienen en el route_path. Y sin usar **!** para no modificar el hash de query_params
          uri.to_s
        end
```